### PR TITLE
Allow proxy server to be configured to respond immediately to CONNECT…

### PR DESCRIPTION
**Pull request by oxtoacart**
… requests

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -32,6 +32,10 @@ type Opts struct {
 	IdleTimeout time.Duration
 	Filter      filters.Filter
 	Dial        proxy.DialFunc
+
+	// OKDoesNotWaitForUpstream can be set to true in order to immediately return
+	// OK to CONNECT requests.
+	OKDoesNotWaitForUpstream bool
```
Comment (oxtoacart) [inline server/server.go]: This is an awkward name, but it allows us to preserve the default behavior when this setting is unspecified.

**Comment by myleshorton** [comment]
LGTM!

**Comment by oxtoacart** [comment]
Thanks @myleshorton!

File: server/server.go (hunk 1 above)
Comment (joesis) [reply server/server.go]: I guess we don't need to keep the name convention of `proxy` package here. Maybe simply `OKImmediately` as the comment suggested?

Comment (oxtoacart) [reply server/server.go]: Yeah, that might sound a little better. @myleshorton already merged this. After we're done with all the PRs, I can just come in and change this one piece.

Comment (joesis) [reply server/server.go]: I didn't notice it's already merged. No big deal anyway.

**Pull request merged**